S3/S3.py

```diff
@@ -701,6 +704,16 @@ def object_put(self, filename, uri, extra_headers = None, extra_label = ""):
             headers['x-amz-server-side-encryption'] = 'aws:kms'
             headers['x-amz-server-side-encryption-aws-kms-key-id'] = self.config.kms_key
 
+        if self.config.sse_customer_key:
+            md5s = md5()
+            sse_customer_key = self.config.sse_customer_key.encode()
+            md5s.update(sse_customer_key)
```

Comment:
(Same for other similar blocks)
You can give the value to hash directly to the md5 like `md5(sse_customer_key)`.
So, I think that it will be cleaner if you pack everything in a single line and avoid temporary vars like:
`b64(md5(key).digest())`

Reply:
Done